### ISSUE TITLE
feature request: haptics on shooting weapon/finishing a reload

### ISSUE THREAD

**Issue by lennyguy20** [LivingFray/HaloCEVR#5]
some simple haptics when firing weapons/finishing reloading/nearby explosions would be really nice for both gameplay (idk why but having haptics just gives weapons so much more oomph in VR) and immersion as well

**Comment by cmibarnwell** [comment]
This is a big ask given how new this mod is - but bHaptics support would be amazing as well. Always on the look out for VR software that works with my TactSuit. Super fun mod here with or without it, just thought to mention it. Thanks for putting this together, hours of fun ahead!

**Comment by RocketMan2007** [comment]
Agreed. I am having an absolute blast with this mod, mad props to the creator. Some haptic feeback in the controllers would be pretty cool.

**Comment by DarkSynopsis** [comment]
Are Haptics there? was just doing the first mission and at one point my brain did go, huh, no haptics when using the assault rifle, since I imagined my hands would be vibrating like crazy but then when I stopped and just did some test firing with it and the pistol I think there is some? its just very feint, unless it was just the pillar exploding or my brain making it up...

**Comment by FolkSong** [comment]
There's a single haptic event when you pull the trigger, so the pistol feels good. But when keeping it held down there's nothing, so automatic weapons don't feel good.

**Comment by benjamincheng10** [comment]
I'm not getting any haptics at all, is there supposed to be or not implemented yet?

**Comment by teddybear082** [comment]
Someone said you can program SteamVR in the SteamVR bindings to trigger a haptic when you hit a button but I haven't tried that.

**Comment by benjamincheng10** [comment]
Since this was the original PC port intended for KB/M, does the game just not have any haptics? Or are haptics support just not activated in the mod yet?

**Comment by TheKrisSodroski** [comment]
I've taken a first stab at this.  

https://github.com/LivingFray/HaloCEVR/compare/master...TheKrisSodroski:HaloCEVR:master

I have weapon resolution and caching working.  Need to play around more to get the final asset names of some of the more obscure weapons.  

Next steps are to implement the haptics here: 

https://github.com/TheKrisSodroski/HaloCEVR/blob/d11369d3013761771f400f2cb4fedfa56d467e00/HaloCEVR/WeaponHandler.cpp#L897

I'll need to play around with the haptic values.  

Haptic api can be found here I think: https://github.com/ValveSoftware/openvr/wiki/SteamVR-Input

Tried to figure out how to get vehicle weapon recognition, but was unable to.  Also tried to see if I could figure out if we could recognize that a grenade went off nearby but didn't get too far.  If anyone has any ideas, feel free to share.
